### DISCUSSION TITLE
Release checklists

### DISCUSSION THREAD

**Discussion by sanket1729** [rust-bitcoin/rust-bitcoin#1106] [Ideas]
I think we should also have a release mini-checklist. So that we don't forget git tags, changelogs/release notes, and (optionally) crev proofs. Feel free to suggest more things if you think something is lacking. Thoughts?

**Comment by Kixunil** [comment]
For breaking releases:

* Did we update all dependencies?
* Is it worth doing semver trick?
* Do we need to write upgrade guidelines?

**Reply by tcharding** [reply]
Where would upgrade guidelines go? I'm also wondering if there is a better place for the new 'policy' doc I'm trying to introduce. Is there any reason not to add a `docs` directory to `rust-bitcoin` and make it, as we do for the github discussions feature, the central place for any org wide docs?

**Reply by Kixunil** [reply]
If it's reasonably short it could be in changelog, if long, we could have a separate file linked in the changelog. `docs` dir sounds good.

**Comment by tcharding** [comment]
Suggestions:

- Step 0: Do the release PR with version bump and changelog
- Step 1: Add a draft PR to miniscript that depends on the release PR branch from step 0 and link to it in the release PR.
- Step 2: Announce the imminent release by discussions announcement to make sure folks get a chance to put forward any open PRs for possible inclusion in the release
- Last Step:  Make discussions announcement for target month for next 'feature freeze' (RC release) six months in the future.

**Comment by tcharding** [comment]
Check all instances of `#[deprecated]` introduced this release include a `since = ...` section with the correct version number.

**Reply by Kixunil** [reply]
Sorry, forgot `since` exists.

**Comment by sanket1729** [comment]
Slightly related checklist: https://github.com/rust-bitcoin/rust-bitcoin/issues/843
